lib/utils.go

```diff
@@ -1,1479 +1,103 @@
-// +build e2e_test
-
-// This is a file with e2e tests for C bindings written in library.go.
-// As a CGO file, it can't have `_test.go` suffix as it's not allowed by Go.
-// At the same time, we don't want this file to be included in the binaries.
-// This is why `e2e_test` tag was introduced. Without it, this file is excluded
-// from the build. Providing this tag will include this file into the build
-// and that's what is done while running e2e tests for `lib/` package.
-
 package main
```

Comment:
We can even split it into `utils.go` with only functions and `types.go` which defines types.

Reply:
Both changes done. 

However, from my point of view, `lib/library.go` should act only as a "router" a bridge between `api/api.go` and `java`, but ATM I see a lot of logic here. May be good to have a closer look at some point.

Reply:
It should but the problem is that some additional logic is required because of differences between CGO and Go types. 

At the same time, it may be more beneficial to look at this from a broader perspective. We have `StatusBackend` which should contain all the logic, we don't really need `StatusAPI` but we need to provide two interfaces: CGO and gomobile. This is for a different swarm though.